app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/MoveNodeFragment.kt

```diff
@@ -0,0 +1,200 @@
+            (parentFragment as? MainFragment)?.putMarkerForCurrentHighlighting(geometry, icon, title)
+        }
+    }
+
+    private fun getPosition(): LatLon? {
+        val createNoteMarker = binding.createNoteMarker
+        val screenPos = createNoteMarker.getLocationInWindow()
+        screenPos.offset(createNoteMarker.width / 2, createNoteMarker.height / 2)
+        return listener?.getMapPositionAt(screenPos)
+    }
+
+    private fun onClickOk() {
+        val pos = getPosition() ?: return
+        if (!checkIsDistanceOkAndUpdateText(pos)) return
+        viewLifecycleScope.launch {
+            val action = MoveNodeAction(pos)
+            elementEditsController.add(editType, node, ElementPointGeometry(node.position), "survey", action)
+            listener?.onMovedNode(editType, pos)
+        }
+    }
+
+    override fun onClickMapAt(position: LatLon, clickAreaSizeInMeters: Double) = false
+
+    @UiThread override fun onMapMoved(position: LatLon) {
+        if (checkIsDistanceOkAndUpdateText(position))
+            okButtonContainer.popIn()
+        else
+            okButtonContainer.popOut()
+    }
+
+    private fun checkIsDistanceOkAndUpdateText(position: LatLon): Boolean {
+        val moveDistance = position.distanceTo(node.position)
+        return when {
+            moveDistance < MIN_MOVE_DISTANCE -> {
+                binding.titleLabel.setText(R.string.node_moved_not_far_enough)
+                false
+            }
+            moveDistance > MAX_MOVE_DISTANCE -> {
+                binding.titleLabel.setText(R.string.node_moved_too_far)
+                false
+            }
+            else -> {
+                binding.titleLabel.text = resources.getString(R.string.node_moved,displayUnit.format(moveDistance.toFloat()))
+                true
+            }
+        }
+    }
+
+    @UiThread override fun onClickClose(onConfirmed: () -> Unit) {
+        if (!hasChanges) {
+            onConfirmed()
+        } else {
+            activity?.let {
+                AlertDialog.Builder(it)
+                    .setMessage(R.string.confirmation_discard_title)
+                    .setPositiveButton(R.string.confirmation_discard_positive) { _, _ ->
+                        onConfirmed()
+                    }
+                    .setNegativeButton(R.string.short_no_answer_on_button, null)
+                    .show()
+            }
+        }
+    }
+
+    companion object {
+        private const val ARG_NODE = "node"
+        private const val ARG_QUEST_TYPE = "quest_type"
+
+        fun create(elementEditType: ElementEditType, node: Node): MoveNodeFragment {
+            val f = MoveNodeFragment()
+            f.arguments = bundleOf(
+                ARG_NODE to Json.encodeToString(node),
+                ARG_QUEST_TYPE to elementEditType.name
+            )
+            return f
+        }
+    }
+}
+
+private const val MIN_MOVE_DISTANCE = 2.0
+private const val MAX_MOVE_DISTANCE = 30.0
```

Comment:
Can you add a rationale (in the comment) why a minimum move distance makes sense and why it is at 2 meters? (The same for max move distance if applicable)

Reply:
I'm not sure about the actual values, that's rather some guess how much makes sense. I added the reasons for limiting the distance.